src/bentoml/_internal/utils/pkg.py

```diff
@@ -35,3 +37,45 @@ def source_locations(pkg: str) -> str | None:
         return
     (module_path,) = module.submodule_search_locations  # type: ignore (unfinished typed)
     return cast(str, module_path)
+
+
+DEFAULT_BENTOML_GIT_URL = "https://github.com/bentoml/bentoml.git"
+
+
+@lru_cache(maxsize=1)
+def get_local_bentoml_dependency() -> str:
+    import logging
```

Comment:
why do we have to create a separate logger here?

Is it because we are running this in a subprocess?

Reply:
hmm how does this work with our workflow, where we have a local bentoml installation without checking it into git yet?

we will still build the wheels, but the version for that case/deps would fail the build process no (given that the rev and url does not exist, yet)

Reply:
>why do we have to create a separate logger here?

It's best practice to create logger with `getLogger(__name__)` for each module, because it will propagate to the parent till `getLogger('bentoml')` if configured properly

>hmm how does this work with our workflow, where we have a local bentoml installation without checking it into git yet?

This is only used by image spec, where we couldn't build wheel for the editable project. This just provide a way to replicate the current bentoml as close as possible.